toolchain/check/BUILD

```diff
@@ -93,6 +91,24 @@ cc_library(
         "//toolchain/sem_ir:typed_insts",
         "@llvm-project//llvm:Support",
     ],
+    alwayslink = 1,
```

Comment:
Leave a comment why this is here?

Reply:
Done

Reply:
Looks like this one was missed, I also wondered if it's actually needed since dump was moved to another cc_library.

Reply:
Oops, I wasn't looking closely enough because I just assumed I'd already removed this set. Removed.